rust/lance-encoding/src/encodings/logical/list.rs

```diff
@@ -113,51 +121,112 @@ impl LogicalPageScheduler for ListPageScheduler {
             let decode_task = scheduled_offsets.drain(num_offsets)?;
             let offsets = decode_task.task.decode()?;
             let numeric_offsets = offsets.as_primitive::<Int32Type>();
-            let start = numeric_offsets.values()[0] as u32;
-            let end = numeric_offsets.values()[numeric_offsets.len() - 1] as u32;
-            trace!(
-                "List offsets range of {:?} maps to item range {:?}..{:?}",
-                range,
-                start,
-                end
-            );
+            // Given ranges [1..3, 5..6] where each list has 10 items we get offsets [[10, 20, 30], [50, 60]]
+            // and we need to normalize to [0, 10, 20, 30]
```

Comment:
Should this say
```suggestion
            // and we need to normalize to [0, 10, 20, 30, 40]
```

since there are 5 elements?

Reply:
Ah, that's the whole reason I created the comment because that tripped me up :).  There are only 3 rows being requested (lists 1, 2, and 5).  So the end result has 4 offsets.  But, in the process of loading it we load 5 offsets because we have to load from two ranges.